server/controllers/assets/utils/transformImage.js

```diff
@@ -0,0 +1,44 @@
+const gm = require('gm');
+const imageMagick = gm.subClass({ imageMagick: true });
+const { getImageHeightAndWidth } = require('../../../utils/imageProcessing');
+
+module.exports = function transformImage(path, queryObj) {
+  return new Promise((resolve, reject) => {
+    let cHeight = queryObj.h || null;
+    let cWidth = queryObj.w || null;
+    let transform = queryObj.t || null;
+    let oHeight,
+      oWidth = null;
+
+    getImageHeightAndWidth(path).then(hwarr => {
```

Comment:
Can use `await`

Reply:
I agree, but that's enough time spent on this feature for now.